src/Xamarin.Android.Build.Tasks/Utilities/JavaObjectsXmlFile.cs

```diff
@@ -0,0 +1,141 @@
+	void ExportJCWTypes (XmlWriter xml)
+	{
+		if (JavaCallableWrappers.Count == 0)
+			return;
+
+		xml.WriteStartElement ("jcw-types");
+
+		XmlExporter.Export (xml, JavaCallableWrappers);
+
+		xml.WriteEndElement ();
+	}
+
+	void ExportACWMappingTypes (XmlWriter xml)
+	{
+		if (ACWMapEntries.Count == 0)
+			return;
+
+		var t = ACWMapEntries.First ();
+
+		xml.WriteStartElement ("acw-types");
+
+		xml.WriteAttributeStringIfNotDefault ("partial-assembly-name", t.PartialAssemblyName);
+		xml.WriteAttributeStringIfNotDefault ("module-name", t.ModuleName);
+
+		foreach (var type in ACWMapEntries) {
+			xml.WriteStartElement ("type");
+			xml.WriteAttributeStringIfNotDefault ("assembly-qualified-name", type.AssemblyQualifiedName);
+			xml.WriteAttributeStringIfNotDefault ("compat-jni-name", type.CompatJniName);
+			xml.WriteAttributeStringIfNotDefault ("java-key", type.JavaKey);
+			xml.WriteAttributeStringIfNotDefault ("managed-key", type.ManagedKey);
+			xml.WriteAttributeStringIfNotDefault ("partial-assembly-qualified-name", type.PartialAssemblyQualifiedName);
+			xml.WriteEndElement ();
+		}
+
+		xml.WriteEndElement ();
+	}
+
+	/// <summary>
+	/// Given an assembly path, return the path to the ".jlo.xml" file that should be next to it.
+	/// </summary>
+	public static string GetJavaObjectsXmlFilePath (string assemblyPath)
+		=> Path.ChangeExtension (assemblyPath, ".jlo.xml");
+
+	public static JavaObjectsXmlFile Import (string filename, JavaObjectsXmlFileReadType readType)
+	{
+		// If the file has zero length, then the assembly wasn't scanned because it couldn't contain JLOs.
+		// This check is much faster than loading and parsing an empty XML file.
+		var fi = new FileInfo (filename);
+
+		if (fi.Length == 0)
+			return unscanned;
+
+		var file = new JavaObjectsXmlFile {
+			WasScanned = true,
+		};
+
+		var xml = XDocument.Load (filename);
+
+		// We let callers specify which part(s) of the file they want to deserialize to save time
+		if (readType.HasFlag (JavaObjectsXmlFileReadType.JCW) && xml.Root?.Element ("jcw-types") is XElement jcw)
+			file.JavaCallableWrappers.AddRange (XmlImporter.Import (jcw));
+
+		if (readType.HasFlag (JavaObjectsXmlFileReadType.ACW) && xml.Root?.Element ("acw-types") is XElement acw) {
+			var partialAssemblyName = acw.GetAttributeOrDefault ("partial-assembly-name", string.Empty);
+			var moduleName = acw.GetAttributeOrDefault ("module-name", string.Empty);
+
+			foreach (var type in acw.Elements ("type")) {
+				var entry = ACWMapEntry.Create (type, partialAssemblyName, moduleName);
+				file.ACWMapEntries.Add (entry);
+			}
+		}
+
+		return file;
+	}
+
+	public static void WriteEmptyFile (string destination, TaskLoggingHelper log)
+	{
+		log.LogDebugMessage ($"Writing empty file '{destination}'");
+
+		// We write a zero byte file to indicate the file couldn't have JLO types and wasn't scanned
+		File.Create (destination).Dispose ();
+	}
+}
+
+[Flags]
+enum JavaObjectsXmlFileReadType
+{
+	None = 0,
+	ACW = 1,
```

Comment:
Could we have "better" names?  To my mind, ACW is "Android Callable Wrapper", JCW is "Java Callable Wrapper", and they're *the same thing*.

Except here.

"ACW" appears to be for the `acw-map.txt` file (https://github.com/xamarin/monodroid/commit/a04b73b30e5bbd81b3773865c313a8084f0a21d1), which is used for Android resource fixup.

Perhaps rename "ACW" to "AndroidResourceFixups"?